src/video/video_source_local.rs

```diff
@@ -221,6 +242,7 @@ impl VideoSource for VideoSourceLocal {
                     }
```

Comment:
You could do on the end here a simple:
```
sizes.sort();
sizes.dedup();
sizes.iter_mut().for_each(|size| size.intervals.sort(); size.intervals.dedup());
```

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=4966b30de796a4491abd7b74d61b0a66

Reply:
Nice, my first thought was with `sort().dedup()`, but I wasn't sure if it was OK to change the list order, although the code intent is way clearear. I'll do that!